cataloguing/additem.pl

```diff
@@ -643,8 +643,13 @@ sub get_common_fields_template {
                     $barcodevalue = $barcodeobj->next_value($oldbarcode) if ($barcodePreference ne 'hbyyyyincr' && ($i > 0 || $exist_itemnumber));
 
                     # Putting it into the record
-                    if ($barcodevalue) {
-                    $record->field($tagfield)->update($tagsubfield => $barcodevalue);
+                    if($barcodevalue) {
+                        $record->field($tagfield)->update( $tagsubfield => $barcodevalue );
+                    }
+                    else {
+                        push @errors, "no_next_barcode";
+                        $itemrecord = $record;
```

Comment:
This is undefined after the loop so not necessary

Reply:
`$record` undefined? It's defined and used to search next barcode

Reply:
ah, it cleans up AFTER loop anyway, yes, I see, so we can't mimic another same way it used when error generated to refill the form with item data

Reply:
because I mimicked how this one with `push @errors, "no_next_barcode";` worked above on the line 616-621, because then later in line 992 if we have itemrecord defined it is used to re-fill form, after ERROR case, for user convenience as I understand.

I can still program some extra logic to not to clean up itemrecord on error, better than remove this line, eh, should I?

Reply:
I don't know how $itemrecord is used, if it is needed in case of error then do the extra logic, if it is not then just remove that line.